main.cpp

```diff
@@ -0,0 +1,76 @@
+#include <iostream>
+#include <gst/gst.h>
+
+#include <viam/sdk/module/module.hpp>
+#include <viam/sdk/module/service.hpp>
+#include <viam/sdk/resource/resource.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/components/camera/server.hpp>
+
+#include "constraints.h"
+#include "csi_camera.h"
+
+using namespace viam::sdk;
+
+int serve(const std::string& socket_path) {
+    // Signal handlers
+    sigset_t sigset;
+    sigemptyset(&sigset);
+    sigaddset(&sigset, SIGINT);
+    sigaddset(&sigset, SIGTERM);
+    pthread_sigmask(SIG_BLOCK, &sigset, NULL);
```

Comment:
I do not think it makes sense to define handlers using std::signal since the "handler" is `server->shutdown()` in the sdk.

We simply block until signal is received then call shutdown in the sdk.

Reply:
>I do not think it makes sense to define handlers using std::signal since the "handler" is server->shutdown() in the sdk

I see. Will resolve.